src/packet/vp9.rs

```diff
@@ -13,6 +13,83 @@ const MAX_VP9REF_PICS: usize = 3;
 /// InitialPictureIDFn is a function that returns random initial picture ID.
 pub type InitialPictureIDFn = Arc<dyn (Fn() -> u16) + Send + Sync + UnwindSafe + RefUnwindSafe>;
 
+/// Vp9 information describing the depacketized/packetized data.
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub struct Vp9CodecExtra {
+    /// Map of the SVC layers.
+    ///
+    /// Index of each element corresponds to `spatial layer` index.
+    ///  
+    /// Each element represents the end (not including) in [`MediaData::data`]
+    /// for `spatial layer` it belongs.
+    ///
+    /// The all `spatial layer`s (up to 3) in [`MediaData::data`] are sorted by
+    /// `spatial layer` index from lower to greater. So, the begining of some
+    /// `spatial layer` is the end (including) of the previous `spatial layer`
+    /// (for the `spatial layer` with index 0 the begining is 0).
+    ///
+    /// ```text
+    /// [MediaData::data]:
+    /// +-----+----------------------+---------------------------+
+    /// | SL0 |         SL1          |            SL2            |
+    /// +-----+----------------------+---------------------------+
+    /// |     |                      |                           |
+    /// V     V                      V                           V
+    /// 0   layers_scheme[0]       layers_scheme[1]            layers_scheme[2]
+    /// ```
+    ///
+    /// [`MediaData::data`]: crate::media::MediaData::data
+    ///
+    /// ## Example
+    ///
+    /// Say you have a VP9 track working in "L3T3" scalability mode and you
+    /// want to retranslate S1T2 layer. So you need to retranslate the all
+    /// spatial and temporal layers <= target layer.
+    ///
+    /// ```no_run
+    /// # use str0m::{format::CodecExtra, media::MediaData};
+    /// #
+    /// // We want to receive S1T2 layer.
+    /// let (target_spatial, target_temporal) = (1u8, 2u8);
+    ///
+    /// let mut media_data: MediaData = todo!();
+    ///
+    /// if let CodecExtra::Vp9(vp9_extra) = media_data.codec_extra {
+    ///     // Try to get current `temporal layer`. If `None` there is no SVC.
+    ///     if let Some(tid) = vp9_extra.tid {
+    ///         if let (true, Some(end_of_target_layer)) = (
+    ///             // Check whether we need this `temporal layer`.
+    ///             tid <= target_temporal,
+    ///             // Check whether we have information about the right
+    ///             // border of data for the target `spatial layer`.
+    ///             vp9_extra.layers_scheme[target_spatial as usize],
+    ///         ) {
+    ///             // Cut off unwanted data of `spatial layer`s with spatial
+    ///             // indeces higher than `target_spatial`.
+    ///             media_data.data.drain(end_of_target_layer..);
```

Comment:
This is a great example!

If someone truly drains the data out he should also update the layer_scheme accordingly ?

Reply:
As far as changing `layers_scheme` does not have any affect on any mechanism and nothing relays on this `layers_scheme` in **str0m** (at least now), we can do anything with it after `MediaData` emitted.
In general, `layers_scheme` field has been added only to let **str0m** clients be able to select SVC layers they need. So, they should decide by themselves what to do with `MediaData::data` and `layers_scheme`.

Reply:
I got you. You mean we should implement some synchronization between `MediaData::data` and `layers_scheme`, right?